src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -197,14 +197,9 @@ protected boolean hasText(final DetailAST slistAST) {
         else {
             final String firstLine = lines[slistLineNo - 1].substring(slistColNo + 1);
             final String lastLine = lines[rcurlyLineNo - 1].substring(0, rcurlyColNo);
-            if (CommonUtils.isBlank(firstLine)
-                    && CommonUtils.isBlank(lastLine)) {
-                // check if all lines are also only whitespace
-                returnValue = !checkIsAllLinesAreWhitespace(lines, slistLineNo, rcurlyLineNo);
-            }
-            else {
-                returnValue = true;
-            }
+            // check if all lines are also only whitespace
+            returnValue = !(CommonUtils.isBlank(firstLine) && CommonUtils.isBlank(lastLine)) ||
+                    !checkIsAllLinesAreWhitespace(lines, slistLineNo, rcurlyLineNo);
```

Comment:
comment is not relevant now.

expression become over complicated to read that is based on negation .... very hard to read.
please revert and use suppression by javadoc tag.